goon-universe/queries.js

```diff
@@ -1,226 +1,626 @@
+var env = require('node-env-file');
+env(__dirname + '/.env');
+
+/*
+ * These constants are concatenated into SQL queries below, so be careful
+ * WHEN IN DOUBT, PATCH CONCATENATION OUT
+ */
+// constants for page numbers/order
+// re-order once all functions are written
+const INTROPAGE = 1
+const INITIAL_REFLECTION = 2
+const CONVERSATION = 3
+const MIDDLE_REFLECTION = 4
+const FINAL_REFLECTION = 5
+const FINAL_ACTION = 6
+const INIT_ACTION = 7
+
+// constants for page types
+const TYPE_PLAIN = 'PLAIN'
+const TYPE_PROMPT = 'PRMPT'
+const TYPE_MCQ = 'MCQ'
+const TYPE_CONV = 'CONV'
+
 const Pool = require('pg').Pool
-require("dotenv").config()
 
 const pool = new Pool({
-  user: process.env.DB_USER,
-  host: process.env.DB_HOST,
-  database: process.env.DB_DATABASE,
-  password: process.env.DB_PASS,
-  port: process.env.DB_PORT,
+    user: env.PGUSER,
+    password: env.PGPASSWORD,
+    host: env.PGHOST,
+    port: env.PGPORT,
+    database: env.PGDATABASE,
+    max: 20,
+    idleTimeoutMillis: 0,
+    connectionTimeoutMillis: 0
 })
 
 function getScenarios(studentID, callback){
-  pool.query('SELECT id, name, description FROM scenarios ORDER BY id ASC', (error, results) => {
-    if (error) {
-      throw error
-    }
-  callback(results.rows)
-  })
+    let thisQuery= 'select scenario.id, scenario.name, scenario.description, scenario.due_date from scenario, partof, enrolled where enrolled.student_id = $1 and enrolled.course_id = partof.course_id and partof.scenario_id = scenario.id '
+    
+    pool.query(thisQuery, [studentID], (error,results) => {
+        if (error) {
+
+            throw error
+        }
+        callback(results.rows)
+    })  
 }
 
-function getIntro(scenarioID, callback){
-  pool.query('SELECT introduction from scenarios where id = $1', [scenarioID], (error,results) => {
-    if(error){
-      throw error
-    }
-    callback(results.rows)
-  })
+function getIntroPage(scenarioID, callback){
+    let thisQuery= 'select plain_page.content from plain_page, pages where pages.id = plain_page.page_id and pages.order = ' + INTROPAGE + 'and scenario_id = $1'
+    
+    pool.query(thisQuery, [scenarioID], (error,results) => {
+        if (error) {
+
+            throw error
+        }
+        callback(results.rows)
+    })  
 }
 
-function getInitReflect(scenarioID, callback){
-  pool.query('SELECT initialreflection from scenarios where id = $1', [scenarioID], (error,results) => {
-    if(error){
-      throw error
-    }
-    callback(results.rows)
-  })
+function getInitReflectPage(scenarioID, callback){
+    let thisQuery = 'select prompt.prompt from pages, prompt where pages.id = prompt.page_id and pages.order = '+ INITIAL_REFLECTION +' and scenario_id = $1'
+    pool.query(thisQuery, [], (error, results) => {
+        if (error){
+            throw error
+        }
+        callback(results.rows)
+    })
 }
```

Comment:
The reflection pages should allow for an arbitrary number of free-response questions, in addition to having some body text. Since Goon Universe's queries didn't allow for that either though, these reflection page queries should be fine to merge in.

Reply:
I believe this should return all prompts that are present on a page. I'll check and reply here if it works

Reply:
confirmed: The SQL query returns all prompts that are part of the initial reflection page. 
By body text do you mean some text that's between any two prompts but not a part of any prompt?

Reply:
I meant text that is separate from the questions, and would display above them.

Reply:
I'm currently working on that in a separate branch.